examples/platforms/posix/flash.c

```diff
@@ -0,0 +1,149 @@
+ *     notice, this list of conditions and the following disclaimer.
+ *  2. Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in the
+ *     documentation and/or other materials provided with the distribution.
+ *  3. Neither the name of the copyright holder nor the
+ *     names of its contributors may be used to endorse or promote products
+ *     derived from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ *  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ *  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ *  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ *  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ *  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <fcntl.h>
+#include <sys/stat.h>
+#include <string.h>
+
+#include <openthread-config.h>
+#include <platform/flash.h>
+
+#include <common/code_utils.hpp>
+#include "platform-posix.h"
+
+static int sFlashFd;
+uint32_t sEraseAddress;
+
+enum
+{
+    FLASH_SIZE = 0x40000,
+    FLASH_PAGE_SIZE = 0x800,
+    FLASH_PAGE_NUM = 128,
+};
+
+ThreadError otPlatFlashInit(void)
+{
+    ThreadError error = kThreadError_None;
+    char fileName[16];
+    struct stat st;
+    bool create = false;
+
+    memset(&st, 0, sizeof(st));
+
+    if (stat("tmp", &st) == -1)
+    {
+        mkdir("tmp", 0777);
+    }
+
+    snprintf(fileName, sizeof(fileName), "tmp/%d.flash", NODE_ID);
+
+    if (access(fileName, 0))
+    {
+        create = true;
+    }
+
+    sFlashFd = open(fileName, O_RDWR | O_CREAT, 0666);
+    lseek(sFlashFd, 0, SEEK_SET);
+
+    VerifyOrExit(sFlashFd >= 0, error = kThreadError_Failed);
+
+    if (create)
+    {
+        for (uint16_t index = 0; index < FLASH_PAGE_NUM; index++)
+        {
+            SuccessOrExit(error = otPlatFlashErasePage(index * FLASH_PAGE_SIZE));
+        }
+    }
+
+exit:
+    return error;
+}
+
+uint32_t otPlatFlashGetSize(void)
+{
+    return FLASH_SIZE;
+}
+
+ThreadError otPlatFlashErasePage(uint32_t aAddress)
+{
+    ThreadError error = kThreadError_None;
+    uint8_t buf = 0xff;
+    uint32_t address;
+
+    VerifyOrExit(sFlashFd >= 0, error = kThreadError_Failed);
+    VerifyOrExit(aAddress < FLASH_SIZE, error = kThreadError_InvalidArgs);
+
+    address = aAddress & (~(uint32_t)(FLASH_PAGE_SIZE - 1));
```

Comment:
While it's clear to me what you are doing here, I'd add a comment explaining the intent of this line.